docs/environment-setup.md

```diff
@@ -107,11 +107,22 @@ _NOTE: This will not work on WSL (Windows Subsystem for Linux). Use the approach
 1. Get [Docker for Desktop For Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac)
 1. Docker for Desktop comes with docker-compose installed.
 
-#### Windows 10 Pro, Enterprise, or Education
+#### Windows 10 Pro, Enterprise, or Education (Hyper-V)
 
 1. Get [Docker for Desktop For Windows](https://hub.docker.com/editions/community/docker-ce-desktop-windows)
 1. Docker for Desktop comes with docker-compose installed.
 
+#### Windows 10 Pro, Enterprise, or Education (Insiders / WSL 2 / Docker)
+
+1. If your Windows build is below 1941, join the [insiders program](https://insider.windows.com/en-us/). Then, update your machine to a newer build through Automatic Updates.
+2. Once installed successfully, install [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/wsl2-install).
+3. Leverage the command `wsl --set-default-version 2`, and install Ubuntu from the Windows Store.
+4. Use `wsl -l -v` to see your Ubuntu instance version, if it's still using WSL version 1, convert it to WSL 2 using `wsl --set-version Ubuntu 2`.
+5. Install The Prerequisties listed on [Docker Desktop WSL Preview](https://docs.docker.com/docker-for-windows/wsl-tech-preview/).
```

Comment:
Since some of these prerequisites (Insider update, WSL) should be installed at this point, I suggest rephrasing "Install The Prerequisties..." to "Install any other prerequisites..."